jpf-symbc/src/main/gov/nasa/jpf/symbc/SymbolicListener.java

```diff
@@ -158,13 +294,58 @@ public void propertyViolated(Search search) {
                 methodSummary = new MethodSummary();
             methodSummary.addPathCondition(pcPair);
             allSummaries.put(currentMethodName, methodSummary);
+            //System.out.println(verifierInfoList.get(0).lineNumber);
+            //System.out.println(verifierInfoList.get(0).returnType);
+            String strPathCondition = pc.toString();
+            //List<Integer> temp = parsePC(strPathCondition);
             System.out.println("Property Violated: PC is " + pc.toString());
             System.out.println("Property Violated: result is  " + error);
             System.out.println("****************************");
+
+
+
+
+            try (BufferedReader reader = new BufferedReader(new FileReader(inputFilePath));
+                 FileWriter writer = new FileWriter(outputFilePath)) {
+                String line;
+                while ((line = reader.readLine()) != null) {
+                    //System.out.println(line);
+                    writer.write(line);
+                    writer.write(System.lineSeparator());
+                }
+                // node part
+                String witnessNode = nodeWriter(verifierInfoList.size(), false);
+                //System.out.println(witnessNode);
+                writer.write(witnessNode);
+                writer.write(System.lineSeparator());
+
+                // edge part, match symbolic variable's value
+                parseSymVar(strPathCondition);
+                String edges = edgeWriter(verifierInfoList.size(), fileName);
+                writer.write(edges);
+                writer.write(System.lineSeparator());
+                // wrap-up the witness file
+                writer.write( System.lineSeparator() + "  </graph>" + System.lineSeparator());
+                writer.write("</graphml>" + System.lineSeparator());
+            } catch (IOException e) {
+                System.out.println("IOException detected: " + e.getMessage());
+                //e.printStackTrace();
+            }
         }
         // }
     }
 
+    boolean interceptSymbolic = false;
```

Comment:
You are not turning this off after we collect the symbolic details. We want to only turn it on once we see Verifier, but then once we collect the needed information, we want to turn it back to false. The process should repeat.

Reply:
I resolved this issue, so I'm closing this review. Thank you.